L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -12,9 +27,16 @@ HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid) const {
   if(TC_id.subdetId()==HGCHEB) {
     return TC_id.cell(); //scintillator
   } else {
+
     int TC_wafer = TC_id.wafer();
-    int TC_12th = ( TC_id.cell() & kSplit_ );
-    return TC_wafer<<kWafer_offset_ | TC_12th;
+    int TC_12th = ( TC_id.cell() & kSplit12_ );
+    int TC_3rd = ( TC_id.cell() & kSplit3_ );
+
+    int thickness = triggerTools_.thicknessIndex(detid,true);
+    int TC_split = TC_12th;
+    if (stcSize_.at(thickness) == kSTCsize16_) TC_split = TC_3rd;
```

Comment:
When you will implement the compatibility with the v9 geometry I think this TC_split choice could be improved with a mapping between STC size and split values:
```
(this is pseudo code)
split_ = {4:0x3a, 16:0x30}
TC_split = TC_id.cell() & split_[stcSize_.at(thickness)]
```
So you don't need to hardcode all the different choices by hand. Everything is coded in a map.

Reply:
Good idea, thanks.